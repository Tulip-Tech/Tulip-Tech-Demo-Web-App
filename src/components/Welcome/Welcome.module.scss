@use 'src/styles/variables';
@use 'src/styles/theme';

.welcome {
  h2 {
    font-weight: var(--body-font-weight-bold);
    font-size: 24px;
  }
}
