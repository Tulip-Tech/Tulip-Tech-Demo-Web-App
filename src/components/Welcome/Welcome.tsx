import React from 'react';
import { useTranslation } from 'react-i18next';

import styles from './Welcome.module.scss';

import Button from '#components/Button/Button';
import useCountdown from '#src/hooks/useCountdown';

type Props = {
  onCloseButtonClick?: () => void;
  onCountdownCompleted?: () => void;
  siteName?: string;
};

const Welcome: React.FC<Props> = ({ onCloseButtonClick, onCountdownCompleted, siteName }) => {
  const { t } = useTranslation('account');
  const countdown = useCountdown(10, 1, onCountdownCompleted);

  return (
    <div className={styles.welcome}>
      <h2>{t('checkout.welcome_title', { siteName })}</h2>
      <p>{t('checkout.welcome_description', { siteName })}</p>
      <Button variant="contained" color="primary" label={t('checkout.start_watching', { countdown })} onClick={onCloseButtonClick} size="large" fullWidth />
    </div>
  );
};

export default Welcome;
