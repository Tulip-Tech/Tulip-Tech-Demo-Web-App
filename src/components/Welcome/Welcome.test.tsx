import React from 'react';
import { act, fireEvent, render } from '@testing-library/react';

import Welcome from './Welcome';

describe('<Welcome>', () => {
  test('renders and matches snapshot', () => {
    const { container } = render(<Welcome siteName="Sitename!" />);

    expect(container).toMatchSnapshot();
  });

  test('calls the onCloseButtonClick callback when clicking the close button', () => {
    const onCloseButtonClick = vi.fn();
    const { getByText } = render(<Welcome onCloseButtonClick={onCloseButtonClick} siteName="Sitename!" />);

    fireEvent.click(getByText('checkout.start_watching'));

    expect(onCloseButtonClick).toBeCalled();
  });

  test('calls the onCloseButtonClick callback when clicking the close button', () => {
    vi.useFakeTimers();
    const onCountdownCompleted = vi.fn();

    render(<Welcome onCountdownCompleted={onCountdownCompleted} siteName="Sitename!" />);

    let i = 10;
    while (i--) {
      act(() => {
        vi.runAllTimers();
      });
    }

    expect(onCountdownCompleted).toBeCalled();
    vi.useRealTimers();
  });
});
