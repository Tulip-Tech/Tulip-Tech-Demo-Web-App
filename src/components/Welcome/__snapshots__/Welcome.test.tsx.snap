// Vitest Snapshot v1

exports[`<Welcome> > renders and matches snapshot 1`] = `
<div>
  <div
    class="_welcome_c4a16f"
  >
    <h2>
      checkout.welcome_title
    </h2>
    <p>
      checkout.welcome_description
    </p>
    <button
      class="_button_f8f296 _primary_f8f296 _fullWidth_f8f296 _large_f8f296"
    >
      <span>
        checkout.start_watching
      </span>
    </button>
  </div>
</div>
`;
